geoapps/utils/string.py

```diff
@@ -0,0 +1,32 @@
+#  Copyright (c) 2022 Mira Geoscience Ltd.
+#
+#  This file is part of geoapps.
+#
+#  geoapps is distributed under the terms and conditions of the MIT License
+#  (see LICENSE file at the root of this source code package).
+
+import os
+import re
+
+
+def existing_file_incrementer(filename):
+    """Add an increment of the form (n) to the filename."""
+
+    if os.path.exists(filename):
+
+        root, file = filename.rsplit(os.path.sep, 1)
+        name, extension = file.split(".", 1)
```

Comment:
This will return the first occurance, which wouldn't work with the time stamp as it is used now.
i.e. `"123.456.789".split(".", 1)` returns `["123", "456.789"]`